### PR TITLE
Turn semicolon as empty statement into an error

### PR DESCRIPTION
**Pull request by dkorpel**
It's been a while: https://github.com/dlang/dmd/pull/6886

### CODE REVIEW

**Review by thewilsonator — APPROVED** [review]
Just double checking this is still allowed in importC

**Review by ljmf00 — APPROVED** [review]
What about the other empty statements? CC @JohanEngelen we have a lot of those.

**Comment by JohanEngelen** [comment]
> What about the other empty statements? CC @JohanEngelen we have a lot of those.

deprecations are already treated as errors, so if this is not adding more deprecations/errors, we're fine.

**Comment by dkorpel** [comment]
> What about the other empty statements?

What other empty statements? `{ }` is allowed.

**Comment by dkorpel** [comment]
> Just double checking this is still allowed in importC

Yes, `cparseStatement` always allowed a semicolon as an empty statement, it never actually checked the `semi` or `semiOk` flag.

**Comment by ljmf00** [comment]
> > What about the other empty statements?
> 
> What other empty statements? `{ }` is allowed.

Sorry, I just got confused with empty declarations :upside_down_face:

**Review by RazvanN7 — APPROVED** [review]

File: changelog/dmd.empty-statement-error.dd [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    foreach (i; 0 .. 8);
+    {
+        // Because of the accidental semicolon above,
+        // this block statement is executed once.
+        // It's not the loop body
+    }
+}
+---
+
+It will now result in an error.
```
Comment (WalterBright) [inline changelog/dmd.empty-statement-error.dd:17]: It always has resulted in an error.

@dkorpel ??

**Pull request merged**